tools/flight_recorder/components/loader.py

```diff
@@ -74,8 +74,14 @@ def read_dir(args: argparse.Namespace) -> Tuple[Dict[str, Dict[str, Any]], str]:
     t0 = time.time()
     version = ""
     filecount = 0
-    assert os.path.isdir(args.folder), f"folder {args.folder} does not exist"
-    for root, _, files in os.walk(args.folder):
+    folder = ""
+    if args.trace_dir:
+        assert os.path.isdir(args.trace_dir), f"folder {args.trace_dir} does not exist"
```

Comment:
Sorry this is the right attribute we should query.. Sorry when doing the refactor, I thought it was `.folder` So let's just use `args.trace_dir`